protocol/messenger.go

```diff
@@ -1337,6 +1338,50 @@ func (m *Messenger) watchExpiredMessages() {
 	}()
 }
 
+// watchUnmutedChats regularly checks for chats that should be unmuted
+func (m *Messenger) watchUnmutedChats() {
+	m.logger.Debug("watching unmuted chats")
+	go func() {
+		for {
+			chats, _ := m.persistence.Chats()
+			select {
+			case <-time.After(3 * time.Second): // Poll every 3 seconds
```

Comment:
Isn't this loop too energy hungry to run every 3s? Especially since it talks to the database to fetch chats. I'm worried about battery consumption for mobile devices.

Reply:
once every 3 seconds is not going to have any noticeable effect on battery life etc, but it should not poll chats from the db, we actually keep them in memory, so you can reach into messenger (`allChats` and loop)

Reply:
If we're going to keep chats in db, Aren't they not going to be updated when the polling happens again ?
And therefore will always be returned

Reply:
we keep chats in the db, you need to save them, but we keep them in memory as well, and we use references, so modifying them in memory will have the effect you want

Reply:
> But what if the list of chats change

It works very much the same as status-mobile, basically we keep all chats in memory, if you change anything, it will be reflected every else, and we just flush to disk every time something changes, but maybe I misunderstand the concern, did I answer the right question :) ?

> I can't say much, never measured this, but I take your word for now ;)

Basically if you think about it, generally we would worry about stuff that happens < 50ms, or busy loops etc, but this is just 3 seconds of inactivity, which is a very long time, almost negligible in computer times, and then some activity which is moderate work, but nothing crazy. There's a cost in creating a new goroutine, but they are fairly cheap as they are green threads, so this is most likely fine. (as a comparison think about a network connection or polling mechanism, that would have to be activated much more often and it would require much more work, but still it's something generally apps handle ok).

Reply:
> It works very much the same as status-mobile, basically we keep all chats in memory, if you change anything, it will be reflected every else, and we just flush to disk every time something changes, but maybe I misunderstand the concern, did I answer the right question :) ?

You did and thank you :)

> Basically if you think about it, generally we would worry about stuff that happens < 50ms

I agree, it should be fine, especially in isolation, the cost might be negligible. I was thinking more in terms of total cost, since the mobile env is quite constrained, so everything adds up. I was also wondering if the mobile OS won't be able to put the process to sleep less frequently, which might affect battery consumption, but I have no clue if that's really a thing to consider, likely not.

It should be fine as you said :+1:, especially because we can take the DB load out of the loop.